src/os/bluestore/BlueStore.h

```diff
@@ -511,59 +533,158 @@ class BlueStore : public ObjectStore,
       ++nref;
     }
     void put() {
-      if (--nref == 0)
-	delete this;
+      if (--nref == 0) {
```

Comment:
The instance will be cleaned up together with the whole `boost::object_pool` instance of `Onode`. However, it seems the refcounting might be completely removed and we can destruct the object earlier, together with a **last** corresponding `Extent` instance. TODO.